src/src/com/microsoft/aad/adal/AuthenticationContext.java

```
@@ -1132,10 +1133,87 @@ private AuthenticationResult acquireTokenLocalCall(final CallbackHandler callbac
+                String base64URLEncodeSignature = URLEncoder.encode(packageHelper.getCurrentSignatureForPackage(mContext.getPackageName()), AuthenticationConstants.ENCODING_UTF8);
+                if(!inputUri.startsWith(AuthenticationConstants.Broker.REDIRECT_PREFIX + "://" + base64URLEncodePackagename + "/"))
+                {
+                    errMsg = "The base64 url encoded package name component of the redirect uri does not match the expected value. "
+                            + "This apps package name is: " + base64URLEncodePackagename
+                            + "so the redirect uri is expected to be:" + actualUri;
+                    exception = new AuthenticationException(ADALError.DEVELOPER_REDIRECTURI_INVALID, errMsg);
+                    Logger.e(TAG + methodName, errMsg , "", ADALError.DEVELOPER_REDIRECTURI_INVALID);     
+                }
+                else if(!inputUri.equalsIgnoreCase(actualUri))
+                {
+                    errMsg = "The base64 url encoded signature component of the redirect uri does not match the expected value. "
+                            + "This apps signature is: " + base64URLEncodeSignature
+                            + "so the redirect uri is expected to be:" + actualUri;
+                    exception = new AuthenticationException(ADALError.DEVELOPER_REDIRECTURI_INVALID, errMsg);
+                    Logger.e(TAG + methodName, errMsg , "", ADALError.DEVELOPER_REDIRECTURI_INVALID);     
+                   
+                }
+            } 
+            catch (UnsupportedEncodingException e) 
+            {
+                exception = new AuthenticationException(ADALError.ENCODING_IS_NOT_SUPPORTED, "The base64 url encoding is not supported."); 
+                Logger.e(TAG + methodName, e.getMessage(), "", ADALError.ENCODING_IS_NOT_SUPPORTED, e);
+            }
+        }
+        
+        if(exception != null)
+        { 
+            throw exception;
```

Why not just re-throw the exception from inside the catch block above.  Then you don't need this check.